### PR TITLE
Add sccache to dockerfile

### PR DESCRIPTION
**Pull request by msadang**
Add provisions to pull stable sccache version and enable it in /usr/bin

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -174,6 +177,8 @@ RUN function comment() { :; }; \
       rm -f /usr/bin/c++; \
       echo "source /etc/cccl.bashrc" >> /etc/bashrc; \
     fi; \
+    curl -SsL ${SCCACHE_URL} | tar -C /usr/bin -zf - --wildcards --strip-components=1 -x */sccache; \
```
Comment (Ethyling) [inline docker/Dockerfile:180]: We should use `${CURL} -L` to be consistent with the other `curl` commands used in this `Dockerfile`

**Review by Ethyling — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (msadang) [reply docker/Dockerfile:180]: Will do.

**Comment by ajschmidt8** [comment]
Please don't merge this until the PR below is reviewed and approved.

- https://github.com/NVIDIA/thrust/pull/1620

**Comment by ajschmidt8** [comment]
Well since the `sccache` builds in https://github.com/NVIDIA/thrust/pull/1620 started without this PR being merged, it seems that the images already contain `sccache`.

**Comment by ajschmidt8** [comment]
After further investigation, it seems that the pull request CI job here was configured to push successful builds to the DockerHub repo. That's why the images included `sccache` already. I've disabled that now.

**Comment by msadang** [comment]
run tests

**Review by allisonvacanti — APPROVED** [review]

**Comment by allisonvacanti** [comment]
LGTM, thanks for helping with this!

**Pull request merged**